app/controllers/api/v2/works_controller.rb

```diff
@@ -0,0 +1,222 @@
+      status = :empty_request
+      messages << "Please provide a list of URLs to find."
+    elsif original_urls.size >= ArchiveConfig.IMPORT_MAX_CHAPTERS
+      status = :too_many_request
+      messages << "Please provide no more than #{ ArchiveConfig.IMPORT_MAX_CHAPTERS } URLs to find."
+    else
+      status = :ok
+      results = find_existing_works(original_urls)
+      messages << "Successfully searched all provided urls"
+    end
+    render_api_response(status, messages, works: results)
+  end
+
+  # POST - create a work and invite authors to claim
+  def create
+    archivist = User.find_by(login: params[:archivist])
+    external_works = params[:items] || params[:works]
+    works_responses = []
+
+    # check for top-level errors (not an archivist, no works...)
+    status, messages = batch_errors(archivist, external_works)
+
+    if status == :ok
+      # Process the works, updating the flags
+      works_responses = external_works.map { |external_work| import_work(archivist, external_work.merge(params.permit!)) }
+      success_works, error_works = works_responses.partition { |r| [:ok, :created, :found].include?(r[:status]) }
+      
+      # Send claim notification emails for successful works
+      if params[:send_claim_emails] && success_works.present?
+        notified_authors = send_external_invites(success_works, archivist)
+        messages << "Claim emails sent to #{notified_authors.to_sentence}."
+      end
+
+      # set final status code and message depending on the flags
+      status = :bad_request if error_works.present?
+      messages = response_message(messages, success_works.present?, error_works.present?)
+    end
+    render_api_response(status, messages, works: works_responses.map { |r| r.except!(:work) })
+  end
+
+  private
+
+  # Set messages based on success and error flags
+  def response_message(messages, any_success, any_errors)
+    messages << if any_success && any_errors
+                  "At least one work was not imported. Please check individual work responses for further information."
+                elsif !any_success && any_errors
+                  "None of the works were imported. Please check individual work responses for further information."
+                else
+                  "All works were successfully imported."
+                end
+    messages
+  end
+
+  # Work-level error handling for requests that are incomplete or too large
+  def work_errors(work)
+    status = :bad_request
+    errors = []
+    urls = work[:chapter_urls]
+    if urls.nil? || urls.empty?
+      status = :empty_request
+      errors << "This work doesn't contain chapter_urls. Works can only be imported from publicly-accessible URLs."
+    elsif urls.length >= ArchiveConfig.IMPORT_MAX_CHAPTERS
+      status = :too_many_requests
+      errors << "This work contains too many chapter URLs. A maximum of #{ArchiveConfig.IMPORT_MAX_CHAPTERS} " \
```

Comment:
This one is definitely consistent with our usual code style, but I noticed in several other places, there are spaces by the brackets, like `#{ ArchiveConfig.IMPORT_MAX_CHAPTERS }`. Could you fix those if you get a chance?